File: pdu/python/trajectory_msgs/pdu_cdr_conv_JointTrajectory.py
from .pdu_pytype_JointTrajectory import JointTrajectory
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..builtin_interfaces.pdu_cdr_conv_Duration import *
from ..std_msgs.pdu_cdr_conv_Header import *
from ..trajectory_msgs.pdu_cdr_conv_JointTrajectoryPoint import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_JointTrajectory(writer: CdrWriter, src: JointTrajectory):
    py_to_cdr_body_Header(writer, src.header)
    writer.write_sequence_length(src.joint_names)
    for elem in src.joint_names:
        writer.write_string(elem)
    writer.write_sequence_length(src.points)
    for elem in src.points:
        py_to_cdr_body_JointTrajectoryPoint(writer, elem)


def cdr_body_to_py_JointTrajectory(reader: CdrReader, dst: JointTrajectory):
    cdr_body_to_py_Header(reader, dst.header)
    dst.joint_names = [reader.read_string() for _ in range(reader.read_uint32())]
    dst.points = []
    for _ in range(reader.read_uint32()):
        elem = JointTrajectoryPoint()
        cdr_body_to_py_JointTrajectoryPoint(reader, elem)
        dst.points.append(elem)
    return dst


def py_to_cdr_JointTrajectory(src: JointTrajectory) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_JointTrajectory(writer, src)
    return writer.bytes()


def cdr_to_py_JointTrajectory(cdr_payload) -> JointTrajectory:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = JointTrajectory()
    return cdr_body_to_py_JointTrajectory(reader, dst)

File: pdu/javascript/trajectory_msgs/pdu_cdr_conv_JointTrajectory.js
import { JointTrajectory } from './pdu_jstype_JointTrajectory.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Duration } from '../builtin_interfaces/pdu_jstype_Duration.js';
import { PduDurationConverter } from '../builtin_interfaces/pdu_cdr_conv_Duration.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { JointTrajectoryPoint } from '../trajectory_msgs/pdu_jstype_JointTrajectoryPoint.js';
import { PduJointTrajectoryPointConverter } from '../trajectory_msgs/pdu_cdr_conv_JointTrajectoryPoint.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduJointTrajectoryConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { JointTrajectory } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        writer.write_sequence_length(src.joint_names);
        for (const elem of src.joint_names) {
            writer.write_string(elem);
        }
        writer.write_sequence_length(src.points);
        for (const elem of src.points) {
            PduJointTrajectoryPointConverter.to_cdr_body(writer, elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { JointTrajectory } dst
     * @returns { JointTrajectory }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        dst.joint_names = Array.from({ length: reader.read_uint32() }, () => reader.read_string());
        dst.points = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.points.push(PduJointTrajectoryPointConverter.cdr_body_to_js(reader, new JointTrajectoryPoint()));
        }
        return dst;
    }

    /**
     * @param { JointTrajectory } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { JointTrajectory }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new JointTrajectory());
    }
}
